### ISSUE TITLE
How to remove a subtree / have children return a list of zippers.

### ISSUE THREAD

**Issue by smucode** [zwilias/elm-rosetree#1]
Hey. What's a good way to remove the currently focused tree? Would it make sense to add a `removeFocused : Zipper a -> Maybe (Zipper a)` that would remove the focused tree and return the parent or root. In case you try to remove the root, it would return `Nothing`.

Also, would it make sense to change `children` to `children : Zipper a -> List (Zipper a)`. I'm traversing my tree by using `children` and in certain cases I need to figure out if the current node has a specific parent.

**Comment by smucode** [comment]
I have a tree which I want to render nicely. To do this I transform the labels and add info about the size and position of each node. I need the children to be calculated first in order to position the parent, so I traverse the tree depth first using `children` and build a new tree.

I have event handlers on some of the nodes, and they require info about if a certain parent is present in the tree. If `children` returned a list of zippers this would be easy to figure out.

Thinking of it I can just as easily collect this info as I traverse down the tree, which probably is the best way to solve this.

What's the reason for `children` returning a list of trees and not zippers btw?

**Comment by zwilias** [comment]
Okay, interesting indeed. I'm going to think about this use-case some more and see if I can come up with a solution!

---

As for `children` returning a list of trees - `children` (like `label`) is one of the "inspect the current focus" functions. So basically, it's just shorthand for `zipper |> Zipper.tree |> Tree.children`.

The API is basically made up like so:
- inspection functions return data from the tree
- navigation functions return a new zipper with a different focus
- modification functions return a new zipper with the same (possibly modified) focus

Since navigating a `Zipper` tends to be a step-by-step process (rather than a way of traversing a tree in on go, for which the raw `Tree` functions should generally offer a nicer API and performance), returning a `List (Zipper a)` would (I think) lead to rather strange results. It could potentially be rather confusing, too - after all, the returned zippers would all be completely independent, and essentially have nothing to do with one another.

Long story short: it feels like a bad API. If someone wants to navigate the children without considering their ancestors, `Zipper.children >> List.map Zipper.fromTree` gives a nice list of non-overlapping zippers. If someone is interested in "does one of my ancestors have label `xyz`", collecting the path to the root is fairly easy: repeat `parent` until it returns `Nothing`!
